src/Filters/stream_filt.jl

```diff
@@ -151,6 +155,47 @@ function FIRFilter(h::Vector, rate::FloatingPoint, Nϕ::Integer=32)
     FIRFilter(kernel, history, historyLen, h)
 end
 
+# Constructor for a resampling FIR filter, where the user needs only to set the sampling rate
+function FIRFilter(rate::FloatingPoint, Nϕ::Integer=32)
+    h = resample_filter(rate, Nϕ)
+    FIRFilter(h, rate)
+end
+
+function FIRFilter(rate::Rational)
+    h = resample_filter(rate)
+    FIRFilter(h, rate)
+end
+
+
+#
+# setphase! set's filter kernel phase index
+#
+function setphase!(kernel::FIRDecimator, ϕ::Real)
+    @assert zero(ϕ) <= ϕ
+    xThrowaway = int(ϕ)
```

Comment:
On 0.4, this should be `round(Int, ϕ)` or `@compat Int(ϕ)`, depending on whether you want to round it or throw an error if it's not exactly convertible to an Int.